### ISSUE TITLE
Question: About development history

### ISSUE THREAD

**Issue by Zhaoyilunnn** [simbricks/gem5#8]
Hi Team,

Thanks for the great work of Simbricks. I am curious about how will this repository be maintained in the future.

Specifically, I found that this repository is forked from a recent (relatively) commit https://github.com/gem5/gem5/commit/bae34876780dfb2bc22b9151bfda1d39ee80cfb1, a version newer than what was used in the simibricks paper.

Will simbricks reimplement adapters each time it plans to integrate a newer version of gem5?

Is it possible to decouple adapters from the original gem5 repository? 

I wish to use the most recent version of gem5 and also wish to use it in simbricks, that's why I ask these above questions.

Thanks,
Yilun

**Comment by jonas-kaufmann** [comment]
Hi @Zhaoyilunnn,

Thanks for your interest. We are very actively developing SimBricks. This on one hand includes improving ease-of-use by introducing better abstractions in our orchestration framework and on the other hand, updating currently supported simulators as well integrating further ones.

gem5 and QEMU are our two core host simulators. The plan is to update them as needed / requested. The SimBricks  adapters are implemented using the simulators' native API. The PCI adapter in gem5, for example, inherits from gem5's `PciDevice` class and purely uses the provided functions. As long as nothing changes fundamentally in the interfaces  or on the Python side where you instantiate your host system, the SimBricks adapters can easily be carried over to a newer gem5 version.

The only thing that has to be done then is to evaluate whether the performance measurementes of the simulated system are still what we expect by, for example, comparing against the experiments in our paper. We are currently working on setting up weekly integration tests as a CI pipeline that check exactly this. Once that is ready, updating gem5 along with SimBricks releases should be straight-forward.

The adapters are also decoupled from the concrete gem5 version in a way. We just rebase our commits on top of the gem5 version we target. We just upgraded gem5 as you pointed out and are still evaluating it. We already found 2 issues that we need to fix before we think of upgrading further:
- https://github.com/simbricks/simbricks/issues/110
- https://github.com/simbricks/simbricks/issues/111

So for the moment, I suggest you rebase yourself. If you have any questions, feel free to ask further questions, for example on our [Slack](https://join.slack.com/t/simbricks/shared_invite/zt-16y96155y-xspnVcm18EUkbUHDcSVonA) for faster responses. It would also be nice if you would let me know whether porting the adapters is straightforward or needs additional work because something changed.
